crypto/ec/ec_pmeth.c

```diff
@@ -103,16 +104,21 @@ static int pkey_ec_sign(EVP_PKEY_CTX *ctx, unsigned char *sig, size_t *siglen,
     unsigned int sltmp;
     EC_PKEY_CTX *dctx = ctx->data;
     EC_KEY *ec = ctx->pkey->pkey.ec;
+    const int sig_sz = ECDSA_size(ec);
 
-    if (!sig) {
-        *siglen = ECDSA_size(ec);
+    /* ensure cast to size_t is safe */
+    if (!ossl_assert(sig_sz > 0))
+        return 0;
+
+    if (sig == NULL) {
+        *siglen = (size_t)sig_sz;
         return 1;
-    } else if (*siglen < (size_t)ECDSA_size(ec)) {
+    } else if (*siglen < (size_t)sig_sz) {
```

Comment:
side note, else after return is jarring, especially because other code flows don't return.